Source/NSFontCollection.m

```diff
@@ -0,0 +1,341 @@
+/* Implementation of class NSFontCollection
+   Copyright (C) 2019 Free Software Foundation, Inc.
+   
+   By: Gregory John Casamento
+   Date: Tue Dec 10 11:51:33 EST 2019
+
+   This file is part of the GNUstep Library.
+   
+   This library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2 of the License, or (at your option) any later version.
+   
+   This library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+   
+   You should have received a copy of the GNU Lesser General Public
+   License along with this library; if not, write to the Free
+   Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+   Boston, MA 02110 USA.
+*/
+
+#import <Foundation/NSArray.h>
+#import <Foundation/NSError.h>
+#import <Foundation/NSValue.h>
+#import <Foundation/NSDictionary.h>
+#import <AppKit/NSFontCollection.h>
+#import <GNUstepGUI/GSFontInfo.h>
+
+// NOTE: Some of this cannot be implemented currently since the backend does not support physically
+//       moving fonts on the filesystem...  this is here for compatilibility for now.
+
+@interface NSMutableFontCollection (Private)
+
+- (void) _setFonts: (NSArray *)fonts;
+- (void) _setQueryAttributes: (NSArray *)queryAttributes;
+
+@end
+
+@implementation NSMutableFontCollection (Private)
+
+- (void) _setFonts: (NSArray *)fonts
+{
+  [_fonts addObjectsFromArray: fonts];
+}
+
+- (void) _setQueryAttributes: (NSArray *)queryAttributes
+{
+  ASSIGN(_queryAttributes, [queryAttributes mutableCopy]);
+}
+
+@end
+
+@interface NSCTFontCollection : NSObject
+@end
+
+@implementation NSCTFontCollection
+@end
+
+@implementation NSFontCollection 
+
+static NSMutableDictionary *__sharedFontCollections;
+static NSMutableDictionary *__sharedFontCollectionsVisibility;
+static NSMutableSet *__sharedFontCollectionsHidden;
+
++ (void) initialize
+{
+  if (self == [NSFontCollection class])
+    {
+      __sharedFontCollections = [[NSMutableDictionary alloc] initWithCapacity: 100];
```

Comment:
Double underscore?

Reply:
https://developer.apple.com/documentation/appkit/nsfontcollection/1497512-showfontcollection?language=objc

There's the proof it manipulates the filesystem.  It returns an NSError from NSFileManager.   Creating GSFontCollections.plist might not be a bad idea.

Double underscore because it's global.

Reply:
Actually interesting thing...

heron@glados FontCollections % pwd                                                                                                                                                                                  [127]
/Users/heron/Library/FontCollections
heron@glados FontCollections %                                                                                                                                                                                        [0]
heron@glados FontCollections % ls                                                                                                                                                                                     [0]
Fixed Width.collection		Modern.collection		Traditional.collection		com.apple.Recents.collection
Fun.collection			PDF.collection			Web.collection

Apparently they are stored as a set of attributes under the Library which means they should have an analogous home in the GNUstep/Library directory.

Reply:
No, this is user not system library.
Looking at the files in my FontCollections directory these are simple archives that store a FontCollection each.

Reply:
Correct, it is the user library.  That's why I mentioned GNUstep/Library.   I simply didn't want to type it each and every time.   If I had meant the system library I would have said /usr/GNUstep/System/Library. :)